src/portable/synopsys/dwc2/dwc2_esp32.h

```diff
@@ -34,10 +34,22 @@
 
 #include "esp_intr_alloc.h"
 #include "soc/periph_defs.h"
-//#include "soc/usb_periph.h"
-
-#define DWC2_REG_BASE       0x60080000UL
-#define DWC2_EP_MAX         6             // USB_OUT_EP_NUM. TODO ESP32Sx only has 5 tx fifo (5 endpoint IN)
+// #include "soc/usb_periph.h"
+
+#if TU_CHECK_MCU(OPT_MCU_ESP32P4)
+  // TODO: ESP32P4 has two USB: USB20 (HS) and USB11 (FS). Make it configurable.
+  #define DWC2_USB20_BASE     0x50000000UL
+  #define DWC2_USB11_BASE     0x50004000UL
+  #define DWC2_REG_BASE       DWC2_USB20_BASE
+  #define DWC2_EP_MAX         8             // USB_OUT_EP_NUM.
```

Comment:
16 for ESP32-P4?
```suggestion
  #define DWC2_EP_MAX         16
```

Reply:
Yep, I see. 
I was distracted by these lines: 

    #define USB_DWC_NUM_EPS                     6
    #define USB_DWC_NUM_IN_EPS                  5  // Todo: Add check for when number of IN channels exceeds limit
    #define USB_DWC_NUM_HOST_CHAN               8

What means: 6 Endpoints, 5 In Endpoints and 8 Channels. 
I was not sure that in case of p4, when we have 16 channels, that means that Endpoints and In Endpoints also would be 16.

Reply:
For P4:
* Num of EPs: 15 (+1 CTRL EP)
* Num of IN EPs: 8 (including 1 CTRL EP)
* Num of host channels: 16 (not relevant for TinyUSB, we do not use their host implementation)